src/server/controllers/match.controller.js

```diff
@@ -51,6 +62,7 @@ export function getMentorList(req, res, next) {
     findConnection(matchOptions.option1, projectOption, localField.mentee),
     findConnection(matchOptions.option2, projectOption, localField.mentor),
     findConnection(matchOptions.option3, projectOption, localField.mentee),
+    saveFilter(req.user._id, req.body.expertise, req.body.career),
```

Comment:
If you add new promise in Promise.all, you should check result's index in next promise. Or just down that line after `User.findOne({ _id: req.user._id }).exec(),`.